core/src/time.rs

```diff
@@ -70,13 +70,9 @@ impl MockClock {
         *current_time = when;
     }
 
-    pub fn advance(&self, dur: &Duration) {
+    pub fn advance(&self, dur: chrono::TimeDelta) {
```

Comment:
Since `Duration` can only represent positive times, the existing version of this method could only move the clock forward, whereas now it can go backward. Should we assert that `dur` is positive, just to catch arithmetic errors elsewhere?

Reply:
84692be994c2b0d8675fb35a969cdbe587dc1ae4